osquery/remote/http_client.h

```diff
@@ -310,6 +310,8 @@ class Client {
 
   void closeSocket();
 
+  void callNetworkOperation(std::function<void()> callback);
```

Comment:
nitpick, can you add a comment for what this function does? Since it does three things, and there are some side effects, such as setting `ec_` can you write an extended docblock with these expectations? Thanks!